wkcuber/image_readers.py

```diff
@@ -148,25 +174,46 @@ def find_count_of_axis(tif_file: TiffFile, axis: str) -> int:
         else:
             return tif_series.shape[index]  # pylint: disable=unsubscriptable-object
 
-    def read_array(self, file_name: str, dtype: np.dtype, z_slice: int) -> np.ndarray:
+    def read_array(
+        self,
+        file_name: str,
+        dtype: np.dtype,
+        z_slice: int,
+        channel_index: Optional[int],
+    ) -> np.ndarray:
         with TiffFile(file_name) as tif_file:
             num_channels = self.read_channel_count(file_name)
+            channel_offset = 0
+            if channel_index is not None:
+                channel_offset = channel_index
+            # if channel_index is not None, then num_channels is one
```

Comment:
Yes, this was a false assumption because I changed it in the `cubing.py` file, but forgot to do it here. After merging the RGB TIFF fix, I reworked and reworded the variables. Please have another look